test/CompressToArchiveTest.php

```diff
@@ -0,0 +1,196 @@
+                [
+                    'adapter' => new Zip(),
+                    'archive' => $basePath . '/archive.zip',
+                ],
+            ],
+            'Tar as String' => [
+                [
+                    'adapter' => 'tar',
+                    'archive' => $basePath . '/archive.tar',
+                ],
+            ],
+            'Tar Instance'  => [
+                [
+                    'adapter' => new Tar(),
+                    'archive' => $basePath . '/archive.tar',
+                ],
+            ],
+        ];
+    }
+
+    /** @param Options $options */
+    #[DataProvider('optionsProvider')]
+    public function testAFilePathWillBeCompressedToTheConfiguredArchive(array $options): void
+    {
+        $filter = new CompressToArchive($options);
+        $path   = __DIR__ . '/Compress/fixtures/directory-to-compress/File1.txt';
+
+        self::assertFileDoesNotExist($options['archive']);
+        $result = $filter->__invoke($path);
+        self::assertFileExists($options['archive']);
+        self::assertSame($options['archive'], $result);
+    }
+
+    /** @param Options $options */
+    #[DataProvider('optionsProvider')]
+    public function testADirectoryWillBeCompressedToTheConfiguredArchive(array $options): void
+    {
+        $filter = new CompressToArchive($options);
+        $path   = __DIR__ . '/Compress/fixtures/directory-to-compress';
+
+        self::assertFileDoesNotExist($options['archive']);
+        $result = $filter->__invoke($path);
+        self::assertFileExists($options['archive']);
+        self::assertSame($options['archive'], $result);
+    }
+
+    /** @param Options $options */
+    #[DataProvider('optionsProvider')]
+    public function testAPHPFileArrayWillYieldACompressedArchive(array $options): void
+    {
+        $filter = new CompressToArchive($options);
+        $path   = __DIR__ . '/Compress/fixtures/directory-to-compress/File1.txt';
+        $files  = [
+            'error'    => UPLOAD_ERR_OK,
+            'tmp_name' => $path,
+            'name'     => 'Foo.txt',
+            'type'     => 'text/plain',
+            'size'     => 0,
+        ];
+
+        self::assertFileDoesNotExist($options['archive']);
+        $result = $filter->__invoke($files);
+        self::assertFileExists($options['archive']);
+        self::assertSame($options['archive'], $result);
+    }
+
+    /** @param Options $options */
+    #[DataProvider('optionsProvider')]
+    public function testPsrUploadedFileCompression(array $options): void
+    {
+        $filter = new CompressToArchive($options);
+        $path   = __DIR__ . '/Compress/fixtures/directory-to-compress/File1.txt';
+        $upload = new UploadedFile(
+            $path,
+            filesize($path),
+            UPLOAD_ERR_OK,
+            'Foo.txt',
+            'text/plain',
+        );
+
+        self::assertFileDoesNotExist($options['archive']);
+        $result = $filter->__invoke($upload);
```

Comment:
If you want to avoid the Laminas\Diactoros\UploadedFile import
```suggestion
        $mockStream = $this->createMock(StreamInterface::class);
        $mockStream->expects(self::once())->method('getMetadata')->willReturn($path);

        $mockFileUpload = $this->createMock(UploadedFileInterface::class);
        $mockFileUpload->expects(self::once())->method('getStream')->willReturn($mockStream);

        self::assertFileDoesNotExist($options['archive']);
        $result = $filter->__invoke($mockFileUpload);
```